### PR TITLE
[Improve][JDBC]jdbcOutputFormat use StatementWrapper

### PR DESCRIPTION
**Pull request by Paddy0523**
## Purpose of this pull request

jdbcOutputFormat use StatementWrapper
1.Use compactionStatementWrapper to precompute on a batch of data, reducing interaction with the database
2.Fix the primary key duplication problem caused by repeatedly modifying the same primary key data in the data restoration job
## Which issue you fix
Fixes # (issue).

## Checklist:

- [ ] I have executed the **'mvn spotless:apply'** command to format my code.
- [ ] I have a meaningful commit message (including the issue id, **the template of commit message is '[label-type-#issue-id][fixed-module] a meaningful commit message.'**)
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have checked my code and corrected any misspellings.
- [ ] My commit is only one. (If there are multiple commits, you can use 'git squash' to compress multiple commits into one.)

### CODE REVIEW

**Comment by PaddyMelody** [comment]
@chaozwn @FlechazoW @ll076110

File: chunjun-connectors/chunjun-connector-jdbc-base/src/main/java/com/dtstack/chunjun/connector/jdbc/dialect/JdbcDialect.java [hunk 1]
```diff
@@ -401,7 +452,7 @@ default KeyUtil<?, BigInteger> initKeyUtil(String incrementName, String incremen
     }
 
     /** catalog/schema/table */
-    default Function<JdbcConf, Tuple3<String, String, String>> getTableIdentify() {
-        return conf -> Tuple3.of(null, conf.getSchema(), conf.getTable());
+    default Tuple3<String, String, String> getTableIdentify(String confSchema, String confTable) {
```
Comment (FlechazoW) [inline chunjun-connectors/chunjun-connector-jdbc-base/src/main/java/com/dtstack/chunjun/connector/jdbc/dialect/JdbcDialect.java:455]: Why 'database' are not in the parameters of function 'getTableIdentity'?

File: chunjun-connectors/chunjun-connector-jdbc-base/src/main/java/com/dtstack/chunjun/connector/jdbc/sink/JdbcDynamicTableSink.java [hunk 2]
```diff
@@ -70,7 +69,6 @@ public ChangelogMode getChangelogMode(ChangelogMode requestedMode) {
                 .addContainedKind(RowKind.INSERT)
                 .addContainedKind(RowKind.DELETE)
                 .addContainedKind(RowKind.UPDATE_AFTER)
-                .addContainedKind(RowKind.UPDATE_BEFORE)
```
Comment (FlechazoW) [inline chunjun-connectors/chunjun-connector-jdbc-base/src/main/java/com/dtstack/chunjun/connector/jdbc/sink/JdbcDynamicTableSink.java:73]: Why delete the 'UPDATE_BEFORE'?

File: chunjun-connectors/chunjun-connector-jdbc-base/src/main/java/com/dtstack/chunjun/connector/jdbc/sink/executor/CachedExecutorProxy.java [hunk 3]
```diff
@@ -0,0 +1,126 @@
+package com.dtstack.chunjun.connector.jdbc.sink.executor;
```
Comment (FlechazoW) [inline chunjun-connectors/chunjun-connector-jdbc-base/src/main/java/com/dtstack/chunjun/connector/jdbc/sink/executor/CachedExecutorProxy.java:1]: New file add license please.

File: chunjun-connectors/chunjun-connector-jdbc-base/src/main/java/com/dtstack/chunjun/connector/jdbc/sink/executor/DynamicSimpleDeleteExecutor.java [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+import java.util.List;
+
+/**
+ * base on row data info to build preparedStatement. row data info include rowkind(which is to set
+ * which sql kind to use )
+ *
+ * <p>Company: www.dtstack.com
+ *
+ * @author xuchao
+ * @date 2021-12-20
```
Comment (FlechazoW) [inline chunjun-connectors/chunjun-connector-jdbc-base/src/main/java/com/dtstack/chunjun/connector/jdbc/sink/executor/DynamicSimpleDeleteExecutor.java:42]: Do not commit any information about the company.

File: chunjun-connectors/chunjun-connector-jdbc-base/src/main/java/com/dtstack/chunjun/connector/jdbc/sink/executor/InsertDeleteBufferReducedExecutor.java [hunk 5]
```diff
@@ -0,0 +1,112 @@
+package com.dtstack.chunjun.connector.jdbc.sink.executor;
```
Comment (FlechazoW) [inline chunjun-connectors/chunjun-connector-jdbc-base/src/main/java/com/dtstack/chunjun/connector/jdbc/sink/executor/InsertDeleteBufferReducedExecutor.java:1]: New file add license please.

File: chunjun-connectors/chunjun-connector-jdbc-base/src/main/java/com/dtstack/chunjun/connector/jdbc/sink/executor/JdbcBatchStatementExecutor.java [hunk 6]
```diff
@@ -0,0 +1,57 @@
+package com.dtstack.chunjun.connector.jdbc.sink.executor;
```
Comment (FlechazoW) [inline chunjun-connectors/chunjun-connector-jdbc-base/src/main/java/com/dtstack/chunjun/connector/jdbc/sink/executor/JdbcBatchStatementExecutor.java:1]: Same as above.

File: chunjun-connectors/chunjun-connector-jdbc-base/src/main/java/com/dtstack/chunjun/connector/jdbc/sink/executor/NoKeyDeleteExecutorProxy.java [hunk 7]
```diff
@@ -0,0 +1,70 @@
+package com.dtstack.chunjun.connector.jdbc.sink.executor;
```
Comment (FlechazoW) [inline chunjun-connectors/chunjun-connector-jdbc-base/src/main/java/com/dtstack/chunjun/connector/jdbc/sink/executor/NoKeyDeleteExecutorProxy.java:1]: Same as above.

File: chunjun-connectors/chunjun-connector-jdbc-base/src/main/java/com/dtstack/chunjun/connector/jdbc/sink/executor/NoKeyInsertDeleteBufferReducedExecutor.java [hunk 8]
```diff
@@ -0,0 +1,106 @@
+package com.dtstack.chunjun.connector.jdbc.sink.executor;
```
Comment (FlechazoW) [inline chunjun-connectors/chunjun-connector-jdbc-base/src/main/java/com/dtstack/chunjun/connector/jdbc/sink/executor/NoKeyInsertDeleteBufferReducedExecutor.java:1]: Same as above.

File: chunjun-connectors/chunjun-connector-jdbc-base/src/main/java/com/dtstack/chunjun/connector/jdbc/sink/executor/RestoreExecutorProxy.java [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+import java.util.concurrent.ExecutionException;
+
+/**
+ * build prepare proxy, proxy implements FieldNamedPreparedStatement. it support to build
+ * preparestmt and manager it with cache.
+ *
+ * <p></>Company: www.dtstack.com
+ *
+ * @author xuchao
+ * @date 2021-12-20
```
Comment (FlechazoW) [inline chunjun-connectors/chunjun-connector-jdbc-base/src/main/java/com/dtstack/chunjun/connector/jdbc/sink/executor/RestoreExecutorProxy.java:40]: Do not commit any information about the company.

File: chunjun-connectors/chunjun-connector-jdbc-base/src/main/java/com/dtstack/chunjun/connector/jdbc/sink/executor/SimpleStatementExecutor.java [hunk 10]
```diff
@@ -0,0 +1,68 @@
+package com.dtstack.chunjun.connector.jdbc.sink.executor;
```
Comment (FlechazoW) [inline chunjun-connectors/chunjun-connector-jdbc-base/src/main/java/com/dtstack/chunjun/connector/jdbc/sink/executor/SimpleStatementExecutor.java:1]: A new file needs to add the license.

File: chunjun-connectors/chunjun-connector-jdbc-base/src/main/java/com/dtstack/chunjun/connector/jdbc/sink/executor/StatementExecutorUtil.java [hunk 11]
```diff
@@ -0,0 +1,249 @@
+package com.dtstack.chunjun.connector.jdbc.sink.executor;
```
Comment (FlechazoW) [inline chunjun-connectors/chunjun-connector-jdbc-base/src/main/java/com/dtstack/chunjun/connector/jdbc/sink/executor/StatementExecutorUtil.java:1]: Same as above.

File: chunjun-connectors/chunjun-connector-jdbc-base/src/main/java/com/dtstack/chunjun/connector/jdbc/sink/executor/UpsertDeleteBufferReducedExecutor.java [hunk 12]
```diff
@@ -0,0 +1,97 @@
+package com.dtstack.chunjun.connector.jdbc.sink.executor;
```
Comment (FlechazoW) [inline chunjun-connectors/chunjun-connector-jdbc-base/src/main/java/com/dtstack/chunjun/connector/jdbc/sink/executor/UpsertDeleteBufferReducedExecutor.java:1]: Same as above.

File: chunjun-core/src/main/java/com/dtstack/chunjun/conf/ContentConf.java [hunk 13]
```diff
@@ -35,7 +35,7 @@ public class ContentConf implements Serializable {
     private TransformerConf transformer;
     private MappingConf nameMapping;
     /** cdc restore conf */
-    private CdcConf restoration = new CdcConf();
```
Comment (FlechazoW) [inline chunjun-core/src/main/java/com/dtstack/chunjun/conf/ContentConf.java:38]: Why remove the default value?

File: chunjun-connectors/chunjun-connector-jdbc-base/src/main/java/com/dtstack/chunjun/connector/jdbc/sink/JdbcDynamicTableSink.java (hunk 2 above)
Comment (PaddyMelody) [reply chunjun-connectors/chunjun-connector-jdbc-base/src/main/java/com/dtstack/chunjun/connector/jdbc/sink/JdbcDynamicTableSink.java:73]: Use upsert statements to handle -U and +U

File: chunjun-connectors/chunjun-connector-jdbc-base/src/main/java/com/dtstack/chunjun/connector/jdbc/dialect/JdbcDialect.java (hunk 1 above)
Comment (PaddyMelody) [reply chunjun-connectors/chunjun-connector-jdbc-base/src/main/java/com/dtstack/chunjun/connector/jdbc/dialect/JdbcDialect.java:455]: image
The Connection object contains only schema and table

**Review by FlechazoW — APPROVED** [review]
LGTM

**Pull request merged**